src/devices/bus/epson_qx/cr1510.cpp

```diff
@@ -0,0 +1,70 @@
+// license:BSD-3-Clause
+// copyright-holders:Brian Johnson
+/*******************************************************************
+ *
+ * Comrex ComFiler CR-1510 Card
+ *
+ *******************************************************************/
+
+#include "cr1510.h"
+#include "emu.h"
+
+
+//**************************************************************************
+//  EPSON CR-1510 DEVICE
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(EPSON_QX_OPTION_CR1510, bus::epson_qx::cr1510_device, "epson_qx_option_cr1510", "Comrex ComFiler CR-1510")
+
+namespace bus::epson_qx {
+
+//-------------------------------------------------
+//  cr1510_device - constructor
+//-------------------------------------------------
+cr1510_device::cr1510_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, EPSON_QX_OPTION_CR1510, tag, owner, clock),
+	device_option_expansion_interface(mconfig, *this),
+	m_hdd(*this, "hdd")
+{
+}
+
+//-------------------------------------------------
+//  device_add_mconfig - device-specific config
+//-------------------------------------------------
+void cr1510_device::device_add_mconfig(machine_config &config)
+{
+	WD1000(config, m_hdd, 0);
+	m_hdd->intrq_wr_callback().set(FUNC(cr1510_device::irq_w));
+
+	HARDDISK(config, "hdd:0", 0);
+}
+
+//-------------------------------------------------
+// IRQ not used on CR-1510
```

Comment:
If unused then why the trampoline?

Reply:
In the wd1000 code the callback for the interrupt uses resolve instead of resolve_safe, this causes this device to seg fault if no interrupt routine is setup as the callback. I suppose thats probably a bug in the wd1000 code so i coul suppit a MR to fix that and then just drop irq handling from this code if thats preferred.

Reply:
Yeah, a PR to fix the wd1000 would be great.

Reply:
Updated this PR with the removal of the unused irq callback. PR #10382 fixes the crash when no callback is set.